backend/src/packages/surveys/survey.entity.ts

```diff
@@ -0,0 +1,99 @@
+import { type Entity } from '#libs/types/types.js';
+
+class SurveyEntity implements Entity {
+  private id: number | null;
+
+  private userId: number | null;
+
+  private preferences: string[];
+
+  private createdAt: Date | null;
+
+  private updatedAt: Date | null;
+
+  private constructor({
+    id,
+    userId,
+    preferences,
+    createdAt,
+    updatedAt,
+  }: {
+    id: number | null;
+    userId: number;
+    preferences: string[];
+    createdAt: string | null;
+    updatedAt: string | null;
+  }) {
+    this.id = id;
+    this.userId = userId;
+    this.preferences = preferences;
+    this.createdAt = createdAt ? new Date(createdAt) : null;
+    this.updatedAt = updatedAt ? new Date(updatedAt) : null;
```

Comment:
We need to pass `Date | null` here and convert this properties outside entity
```suggestion
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
```